grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/ArmeriaClientCall.java

```diff
@@ -472,6 +500,30 @@ private void prepareHeaders(Compressor compressor, Metadata metadata, long remai
         ctx.updateRequest(newReq);
     }
 
+    private void closeWhenListenerThrows(Throwable t) {
+        if (needsDirectInvocation()) {
+            close(GrpcStatus.fromThrowable(t), new Metadata());
+        } else {
+            execute(() -> close(GrpcStatus.fromThrowable(t), new Metadata()));
+        }
```

Comment:
Minor suggestion for reducing duplication:

```suggestion
        closeWhenEos(GrpcStatus.fromThrowable(t), new Metadata());
```

Reply:
Nice, ty!